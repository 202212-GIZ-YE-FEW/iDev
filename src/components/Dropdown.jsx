import clsx from "clsx";
import Image from "next/image";
import Link from "next/link";
import { useAuth } from "./context/AuthContext";
function Dropdown(prop) {
    const { Logout } = useAuth();
    const { setOpenDropdown, openDropdown, icon, mobile = false, links } = prop;
    return (
        <div
            className='relative z-20'
            onClick={() => {
                setOpenDropdown(!openDropdown);
            }}
        >
            {icon}
            {openDropdown ? (
                <div
                    className={clsx(
                        "w-48 py-2 mt-2 origin-top-right bg-white rounded-md shadow-xl",
                        {
                            "absolute start-0 z-20": mobile === false,
                        }
                    )}
                >
                    {links.map((link) => {
                        return link.name === "logout" ? (
                            <Link
                                key={link.name}
                                href=''
                                onClick={Logout}
                                className='block px-4 py-3 text-sm md:text:lg text-gray font-medium capitalize transition-colors duration-300 transform hover:bg-cyan'
                            >
                                {link.name}
                            </Link>
                        ) : (
                            <Link
                                key={link.name}
                                href={link.href}
                                className='block px-4 py-3 text-sm md:text:lg text-gray font-medium capitalize transition-colors duration-300 transform hover:bg-cyan'
                            >
                                {link.name}
                            </Link>
                        );
                    })}
                </div>
            ) : (
                ""
            )}
        </div>
    );
}
export default Dropdown;
