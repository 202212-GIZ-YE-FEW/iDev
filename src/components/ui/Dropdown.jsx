function Dropdown(props) {
    const {
        label,
        placeholder,
        value,
        data,
        // FontSize = "3xl",
        FontWeight = "light",
        WidthSize = "w-full",
        radius = "md",
    } = props;
    const inputClasses = `${WidthSize} border border-solid self-center border-gray-300 px-[20px] py-[10px] min-w-0 text-sm font-${FontWeight} rounded-${radius}`;

    const handleChange = (event) => {
        event.target;
        // onChange(value);
    };
    return (
        <>
            {/* {label && (
                <label className='font-normal   whitespace-nowrap flex:me-10 flex:self-center'>
                    {label}
                </label>
            )} */}

            <select
                value={value}
                onChange={handleChange}
                placeholder={placeholder}
                className={inputClasses}
            >
                {data?.map((item, key) => {
                    return (
                        <option key={key} value={item.value}>
                            {item.lable}
                        </option>
                    );
                })}
            </select>
        </>
    );
}

export default Dropdown;
