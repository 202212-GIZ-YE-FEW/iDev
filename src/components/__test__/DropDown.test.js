import renderer from "react-test-renderer";

import Dropdown from "@/components/Dropdown/Dropdown";

it("renders correctly", () => {
    const tree = renderer.create(<Dropdown />).toJSON();
    expect(tree).toMatchSnapshot();
});
