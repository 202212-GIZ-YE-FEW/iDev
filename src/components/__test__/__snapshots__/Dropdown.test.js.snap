// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="relative z-20"
  onClick={[Function]}
>
  <div
    className="w-48 py-2 mt-2 origin-top-right bg-white rounded-md shadow-xl absolute absolute end-0 z-20"
  />
</div>
`;
